### PR TITLE
RTL: For VTOL first descend before transition

### PR DESCRIPTION
**Pull request by RomanBapst**
When there is no planned mission landing available which makes the vehicle reduce altitude before transitioning then the vehicle will transition back to hover at the RTL return altitude as soon as it reaches the home position.
The RTL return altitude can be quite high (imagine the vehicle needs to clear a hill).
In most cases it's not desired for the vehicle to transition at high altitudes as hover flight uses a lot of energy. It would be better for the vehicle to descend in fixed wing mode and only transition once it's at the RTL descend altitude.
Now that we have path tracking during the transition it's actually ok to loiter down to RTL descend altitude and then transition. Previous concerns have always been that the vehicle drifts off during the transition as it was done in an open loop fashion.

### CODE REVIEW

File: src/modules/navigator/rtl.cpp [hunk 1]
```diff
@@ -318,7 +318,13 @@ RTL::set_rtl_item()
 		}
 
 	case RTL_STATE_DESCEND: {
-			_mission_item.nav_cmd = NAV_CMD_WAYPOINT;
+			if (_navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING) {
+				_mission_item.nav_cmd = NAV_CMD_LOITER_TO_ALT;
```
Comment (RomanBapst) [inline src/modules/navigator/rtl.cpp:322]: @dagar For some reason this does not work for plain multirotors.

**Comment by xdwgood** [comment]
@RomanBapst  

> path tracking during the transition

Where is this function implemented? Does tailsitter also have this feature? thanks

File: src/modules/navigator/rtl.cpp (hunk 1 above)
Comment (dayjaby) [reply src/modules/navigator/rtl.cpp:322]: Can you try to set a loiter radius? There is probably the same issue as in https://github.com/PX4/Firmware/pull/14742

Comment (dagar) [reply src/modules/navigator/rtl.cpp:322]: The `LOITER_TO_ALT` logic was only implemented for FW. 

https://github.com/PX4/Firmware/blob/d0dcbae4af7e0aa993d0fb48f9da0becc4a663ae/src/modules/navigator/mission_block.cpp#L236-L237

**Comment by sfuhrer** [comment]
This is how it looks like now (with this PR): 
1) The vehicle returns on RTL_RETURN_ALT,
2) then begins FW loiter and descends to RTL_DESCEND_ALT.
3) Once that is reached, it transitions to hover mode (while still tracking the loiter),
4) and then moves over the land spot in hover mode, still on RTL_DESCEND_ALT. 
5) Then, if  RTL_LAND_DELAY is set, it will wait there for this amount of time, 
6) and proceeds with landing afterwards.
image

**Comment by hamishwillee** [comment]
@sfuhrer OK, so docs for return mode are here: http://docs.px4.io/master/en/flight_modes/return.html#vtol

The current instructions basically say "If not in a mission landing, a VTOL in FW mode will always transition back to MC just before landing (ignoring NAV_FORCE_VT)."

So the difference now appears to be that it does the transition explicitly at RTL_DESCEND_ALT. 
1. Does it still ignore NAV_FORCE_VT ?

You don't mention the behaviour around `RTL_CONE_ANG` - I assume it is the same as for MC? I.e. the cone angle defines the _effective_ RTL_RETURN_ALT, but otherwise things are the same.

Further, the rest of the doc doesn't talk about transitions for VTOL much, but my assumption is that this really only applies for the cases where landing is not defined by a mission right (i.e. not when the return mode is to follow a mission path).

**Comment by VTOLDavid** [comment]
Very nice feature. Would be a good point to add wind direction into the procedure in order to fly with headwind from the end of the transition to the landing point.

**Comment by LorenzMeier** [comment]
@sfuhrer Is this still current?

**Comment by sfuhrer** [comment]
> @sfuhrer Is this still current?

No, replaced by https://github.com/PX4/PX4-Autopilot/pull/16377.

**Comment by VTOLDavid** [comment]
> 
> 
> @RomanBapst
> 
> > path tracking during the transition
> 
> Does tailsitter also have this feature? thanks

I think they do, and I having some dangerous transitions because of this, would be nice if it could be disabled.

**Comment by VTOLDavid** [comment]
> 
> 
> @RomanBapst
> 
> > path tracking during the transition
> 
> Does tailsitter also have this feature? thanks

@RomanBapst 
I think they do, and I having some dangerous transitions because of this, would be nice if it could be disabled.

**Pull request closed without merging**